SAK-40103 Add event tracking to the GradebookNG tool

payten opened the pull request:
Delivers https://jira.sakaiproject.org/browse/SAK-40103.

This PR aims to add many of the events previously generated by Gradebook Classes to GradebookNG.  It also adds a few new events based on some of the new features available in the tool.

The `EventHelper` serialises the data to match the `ref` originally generated by Gradebook Classic where possible.

mpellicer commented:
Looks good to me Payten, thanks for this feature.

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GbRole.java:
```diff
@@ -41,4 +41,9 @@
 		return this.value;
 	}
+
+	public String toEventString() {
```
is this meant to be in this class? roles with events?

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:
```diff
@@ -778,6 +779,13 @@ public GradeSaveResponse saveGrade(final Long assignmentId, final String student
+		try {
+			EventHelper.postUpdateGradeEvent(gradebook, assignmentId, studentUuid, newGrade, rval, getUserRole(getCurrentSiteId()));
+		} catch (GbAccessDeniedException e) {
+			// oh well
```
this will cause a sonar error, can you please log.debug this if its not an issue that an event doesn't happen. is anything relying on events and what would cause a failure?

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:
```diff
@@ -1697,6 +1705,12 @@ public Long addAssignment(final Assignment assignment) {
+			try {
+				EventHelper.postAddAssignmentEvent(gradebook, assignmentId, assignment, getUserRole(getCurrentSiteId()));
+			} catch (GbAccessDeniedException e) {
+				// oh well
```
per above

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:
```diff
@@ -1940,11 +1954,19 @@ public boolean updateAssignment(final Assignment assignment) {
+			try {
+				EventHelper.postUpdateAssignmentEvent(gradebook, assignment, getUserRole(getCurrentSiteId()));
+			} catch (GbAccessDeniedException e) {
+				// oh well
```
and this one

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:
```diff
@@ -2009,6 +2031,13 @@ public boolean updateUngradedItems(final long assignmentId, final double grade,
+			try {
+				EventHelper.postUpdateUngradedEvent(gradebook, assignmentId, String.valueOf(grade), getUserRole(getCurrentSiteId()));
+			} catch (GbAccessDeniedException e) {
+				// oh well
```
tagging this as per above

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:
```diff
@@ -2104,6 +2133,13 @@ public boolean updateAssignmentGradeComment(final long assignmentId, final Strin
+			try {
+				EventHelper.postUpdateCommentEvent(getGradebook(), assignmentId, studentUuid, comment, getUserRole(getCurrentSiteId()));
+			} catch (GbAccessDeniedException e) {
+				// oh well
```
:)

steveswinsburg commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:
```diff
@@ -2262,6 +2300,12 @@ public void updateGradebookSettings(final GradebookInformation settings) {
+		try {
+			EventHelper.postDeleteAssignmentEvent(getGradebook(), assignmentId, getUserRole(getCurrentSiteId()));
+		} catch (GbAccessDeniedException e) {
+			// oh well
```
one more

payten commented:
Thanks for the review @steveswinsburg.  I pushed up a patch that cleans those things up!

Namely, I've removed the `GbRole.toEventString` and introduced a `getUserRoleOrNone` method which returns the user's GbRole or `GbRole.NONE`.  We can now generate an event with a somewhat meaningful role string in cases where the user does not have access to the site (probably never happen) or if their role is not supported by the tool (possible but unlikely?) -- this was just me trying to replicate the Gradebook Classic `ref` values which included that role value.
